metpy/plots/station_plot.py

```diff
@@ -59,8 +59,8 @@ def __init__(self, ax, x, y, fontsize=10, spacing=None, transform=None, **kwargs
 
         """
         self.ax = ax
-        self.x = x
-        self.y = y
+        self.x = np.array(x)
```

Comment:
Thinking about this further, I'm uncomfortable with us writing code that I *know* will strip units unless we need to. Can you replace with `atleast_1d` from `metpy.units`, which should accomplish the same thing but ensure that we keep units if they're present.